core/src/main/java/discord4j/core/spec/GuildMemberModifySpec.java

```diff
@@ -0,0 +1,42 @@
+package discord4j.core.spec;
+
+import discord4j.core.object.util.Snowflake;
+import discord4j.rest.json.request.GuildMemberModifyRequest;
+
+import javax.annotation.Nullable;
+
+public class GuildMemberModifySpec implements Spec<GuildMemberModifyRequest> {
+
+    private final GuildMemberModifyRequest.Builder builder = GuildMemberModifyRequest.builder();
+
+    public GuildMemberModifySpec setNewVoiceChannel(@Nullable Snowflake channel) {
+        builder.channelId(channel == null ? null : channel.asLong());
+        return this;
+    }
+
+    public GuildMemberModifySpec setMute(boolean mute) {
+        builder.mute(mute);
+        return this;
+    }
+
+    public GuildMemberModifySpec setDeafen(boolean deaf) {
+        builder.deaf(deaf);
+        return this;
+    }
+
+    public GuildMemberModifySpec setNickname(@Nullable String nickname) {
+        builder.nick(nickname == null ? "" : nickname);
+        return this;
+    }
+
+    public GuildMemberModifySpec setRoles(@Nullable long[] roles) {
```

Comment:
Is there any other spec that uses an array of IDs? If not, then I think the precedence for this should be `Set<Snowflake>`. If there is and it uses this format then I guess we should change those to that.